it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/bigtable/BigtableResourceManager.java

```diff
@@ -357,6 +359,20 @@ public synchronized void createTable(String tableId, BigtableTableSpec bigtableT
           cdcEnabledTables.add(tableId);
         }
         tableAdminClient.createTable(createTableRequest);
+
+        boolean allReady;
+        do {
+          allReady = true;
+          Map<String, ReplicationState> states = tableAdminClient.getTable(tableId)
+              .getReplicationStatesByClusterId();
+          for (ReplicationState state : states.values()) {
+            if (!state.equals(ReplicationState.READY)) {
+              allReady = false;
+              Thread.sleep(5000);
+              break;
+            }
+          }
+        } while (!allReady);
```

Comment:
nit: I would use Awaitility to do thread waiting instead of calling Thread.sleep (adjusting the Durations as needed)
```suggestion
        Awaitility.await("Waiting for all tables to be replicated.")
        .atMost(Duration.ofMinutes(1))
        .pollInterval(Duration.ofMillis(500))
        .until(() -> {
            tableAdminClient.getTable(tableId)
                .getReplicationStatesByClusterId()
                .values()
                .stream()
                .allMatch(Table.ReplicationState.READY::equals)
        });
```